e2e/cypress/integration/ui/infra-proxy/chef-servers.spec.ts

```diff
@@ -0,0 +1,127 @@
+describe('chef server', () => {
+  const now = Cypress.moment().format('MMDDYYhhmm');
+  const cypressPrefix = 'infra';
+  const serverName = `${cypressPrefix} server ${now}`;
+  const generatedServerID = serverName.split(' ').join('-');
+  const customServerID = `${cypressPrefix}-custom-id-${now}`;
+  const serverFQDN = 'chef-server-1617089723092818000.com';
+  const serverIP = '176.119.50.159';
+
+  before(() => {
+    cy.adminLogin('/infrastructure/chef-servers').then(() => {
+      const admin = JSON.parse(<string>localStorage.getItem('chef-automate-user'));
+      cy.get('app-welcome-modal').invoke('hide');
+    });
+    cy.restoreStorage();
+  });
+
+  beforeEach(() => {
+    cy.restoreStorage();
+  });
+
+  afterEach(() => {
+    cy.saveStorage();
+  });
+
+  describe('chef server list page', () => {
+
+    it('can add a server', () => {
+      cy.get('[data-cy=add-server-button]').contains('Add Chef Server').click();
+      cy.get('app-chef-servers-list chef-modal').should('exist');
+      cy.get('[data-cy=add-name]').type(serverName);
+      cy.get('[data-cy=id-label]').contains(generatedServerID);
+      cy.get('[data-cy=add-fqdn]').type(serverFQDN);
+      cy.get('[data-cy=add-ip-address]').type(serverIP);
+
+      cy.get('[data-cy=add-button]').click();
+      cy.get('app-chef-servers-list chef-modal').should('not.be.visible');
+
+      // verify success notification and then dismiss it
+      // so it doesn't get in the way of subsequent interactions
+      cy.get('app-notification.info').should('be.visible');
+      cy.get('app-notification.info chef-icon').click();
+
+      cy.contains(serverName).should('exist');
+    });
+
+    it('lists servers', () => {
+      cy.get('[data-cy=add-server-button]').contains('Add Chef Server');
+
+      cy.get('#servers-table-container chef-th').contains('Name');
+      cy.get('#servers-table-container chef-th').contains('FQDN');
+      cy.get('#servers-table-container chef-th').contains('IP Address');
+      cy.get('#servers-table-container chef-th').contains('Number Of Orgs');
+    });
+
+    it('can create a chef server with a custom ID', () => {
+      cy.get('[data-cy=add-server-button]').contains('Add Chef Server').click();
+      cy.get('app-chef-servers-list chef-modal').should('exist');
+      cy.get('[data-cy=add-name]').type(serverName);
+      cy.get('[data-cy=add-id]').should('not.be.visible');
+      cy.get('[data-cy=edit-button]').contains('Edit ID').click();
+      cy.get('[data-cy=id-label]').should('not.be.visible');
+      cy.get('[data-cy=add-id]').should('be.visible').clear().type(customServerID);
+      cy.get('[data-cy=add-fqdn]').type(serverFQDN);
```

Comment:
Simply a suggestion - thinking it _might_ be useful if there was a check halfway through that the "Create Server" button is `disabled` since the form isn't fully filled out yet.

Maybe this suggestion isn't helpful - so do with it what you will 🤷

Reply:
Yes, good suggestion! I would put the check just before the last input, confirming that it is disabled until all inputs are filled in. The click (L67) already does an implicit confirmation that it becomes enabled (but if you want to add an explicit check for enabled, that is OK too).